apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget_tick.js

```diff
@@ -0,0 +1,72 @@
+/* global Stream, DOMEventSource, MinuteClockSource */
+/* global LockScreenClockWidgetSuspend, LockScreenBasicState */
+'use strict';
+
+/***/
+(function(exports) {
+  var LockScreenClockWidgetTick = function(component) {
+    LockScreenBasicState.apply(this, arguments);
+    this.configs.name = 'LockScreenClockWidgetTick';
+    this.configs.stream.events = [
+      'ftudone',
+      'moztimechange',
+      'lockscreen-notification-clock-tick'
+    ];
+    this.configs.stream.interrupts = [
+      'screenchange'
+    ];
+    this._minuteSource = new MinuteClockSource({
+      'type': 'lockscreen-notification-clock-tick'
+    });
+
+    this.configs.stream.sources =
+      [new DOMEventSource({events: [
+        'screenchange',
+        'ftudone',
+        'moztimechange',
+        'screenchange',
+        'lockscreen-notification-clock-tick'
+        ]}),
+        this._minuteSource
+      ];
+    this.handleSourceEvent = this.handleSourceEvent.bind(this);
+  };
+  LockScreenClockWidgetTick.prototype =
+    Object.create(LockScreenBasicState.prototype);
+
+  LockScreenClockWidgetTick.prototype.start = function() {
+    this.stream = new Stream(this.configs.stream);
+    return this.stream.start(this.handleSourceEvent)
+      .next(this.component.updateClock.bind(this.component))
+      .next(this.stream.ready.bind(this.stream));
+  };
+
+  LockScreenClockWidgetTick.prototype.stop =
+  function() {
+    return LockScreenBasicState.prototype.stop.call(this)
+      .next(this._minuteSource.stop.bind(this._minuteSource));
```

Comment:
Why would you need to explicitly stop this source? I thought it will be all done automatically underneath?